CMakeLists.txt

```
@@ -74,6 +74,7 @@ CHECK_MINIMAL_CXX_STANDARD(11 ENFORCE)
 # Project dependencies
 ADD_PROJECT_DEPENDENCY(pinocchio 2.3.1 REQUIRED)
 ADD_PROJECT_DEPENDENCY(eiquadprog 1.1.3 REQUIRED)
+FIND_PACKAGE(qpmad QUIET)
```

```suggestion
ADD_PROJECT_DEPENDENCY(qpmad QUIET)
```
Because the following `TARGET_INCLUDE_DIRECTORIES` is declared as `PUBLIC` and the included (and installed)  header leaks the dependency to qpmad

---

Using cmake 3.22

in case qpmad is not found, ```ADD_PROJECT_DEPENDENCY``` leads to a cmake find_package error in libraries that links to tsid.

the [find_dependency](https://cmake.org/cmake/help/v3.13/module/CMakeFindDependencyMacro.html?highlight=find_dependency#command:find_dependency) command inside the tsid cmake config file will fail, even with the ```QUIET``` keyboard.

My understanding is that ```find_dependency``` should not be used for optional components, and qpmad is supposed to be optional.

With find_package this problem does not occur. The ```TARGET_INCLUDE_DIRECTORIES``` is already computed only in case qpmad is found and installed in the system.

---

You are right, I didn't think about this case. 

But if we don't want to leak the dependency, I think we should use a `PRIVATE` option for the `TARGET_INCLUDE_DIRECTORIES`, and maybe avoid installing publicly this header, which include a qpmad header.

Otherwise, a workaround would be to keep this `find_package`,and add a `ADD_PROJECT_DEPENDENCY` inside a ` if(qpmad_FOUND)` block. 

What do you think ?

---

Using the ```PRIVATE``` option will force anyone that want to use this solver with tsid to ```find_package``` again manually to be able to find the correct headers.

I prefer the second way.

The header are installed only in case qpmad is found.